src/application/shell.ts

```diff
@@ -341,6 +386,60 @@ class ApplicationShell extends Widget {
     this._rightHandler.sideBar.currentChanged.connect(this._save, this);
   }
 
+  /*
+   * Return the TabBar that has the currently active Widget or undefined.
+   */
+  private _currentTabBar(): TabBar {
+    let current = this._dockPanel.currentWidget;
+    if (current) {
+      let title = current.title;
+      let tabBar = find(this._dockPanel.tabBars(), bar => contains(bar.titles, title));
+      return tabBar
+    }
+    return void 0;
+  }
+
+  /*
+   * Return the TabBar previous to the current TabBar (see above) or undefined.
+   */
+  private _previousTabBar(): TabBar {
+    let current = this._currentTabBar();
+    if (current) {
+      let bars = toArray<TabBar>(this._dockPanel.tabBars());
```

Comment:
the `<TabBar>` type is not needed. `toArray` will automatically infer the generic type based on the argument.